src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -823,6 +823,8 @@ private bool WriteLocalFileHeader(bool isEmptyFile)
                 }
                 else // if we are not in streaming mode, we have to decide if we want to write zip64 headers
                 {
+                    // When updating and when we can seek, the data descriptor bit needs to be flipped back to prevent corruption
+                    _generalPurposeBitFlag &= ~BitFlagValues.DataDescriptor;
```

Comment:
Possibly `WriteDataDescriptor()` could be improved to assert that `_generalPurposeBitFlag & BitFlagValues.DataDescriptor != 0`

Reply:
if the tests pass with that in place, yo ucould add it.

Reply:
Ok thanks, I see what you mean. I can add this assert to WriteDataDescriptor.

Reply:
I forgot to address this in iteration 2. I'll add the assert in iteration 3.

Reply:
Done. See below.